src/client.rs

```diff
@@ -0,0 +1,237 @@
+    pub async fn new(cfg: Option<QdrantClientConfig>) -> Result<Self> {
+        let cfg = cfg.unwrap_or_default();
+
+        let endpoint = Channel::builder(cfg.uri.parse().unwrap())
+            .timeout(cfg.timeout)
+            .connect_timeout(cfg.connect_timeout)
+            .keep_alive_while_idle(cfg.keep_alive_while_idle);
+        let channel = endpoint.connect().await?;
+
+        let collection_api = CollectionsClient::new(channel.clone());
+        let points_api = PointsClient::new(channel);
+
+        let client = Self {
+            collection_api,
+            points_api,
+        };
+
+        Ok(client)
+    }
+
+    pub async fn has_collection(&mut self, collection_name: impl ToString) -> Result<bool> {
+        let result = self
+            .collection_info(collection_name)
+            .await?
+            .result
+            .is_some();
+        Ok(result)
+    }
+
+    pub async fn create_collection(
+        &mut self,
+        details: CreateCollection,
+    ) -> Result<CollectionOperationResponse> {
+        let result = self.collection_api.create(details).await?;
+        Ok(result.into_inner())
+    }
+
+    pub async fn delete_collection(
+        &mut self,
+        collection_name: impl ToString,
+    ) -> Result<CollectionOperationResponse> {
+        let result = self
+            .collection_api
+            .delete(DeleteCollection {
+                collection_name: collection_name.to_string(),
+                ..Default::default()
+            })
+            .await?;
+        Ok(result.into_inner())
+    }
+
+    pub async fn collection_info(
+        &mut self,
+        collection_name: impl ToString,
+    ) -> Result<GetCollectionInfoResponse> {
+        let result = self
+            .collection_api
+            .get(GetCollectionInfoRequest {
+                collection_name: collection_name.to_string(),
+            })
+            .await?;
+        Ok(result.into_inner())
+    }
+
+    pub async fn upsert(
+        &mut self,
+        collection_name: impl ToString,
+        points: Vec<Point>,
+    ) -> Result<PointsOperationResponse> {
+        self._upsert(collection_name, points, false).await
+    }
+
+    pub async fn upsert_blocking(
+        &mut self,
+        collection_name: impl ToString,
+        points: Vec<Point>,
+    ) -> Result<PointsOperationResponse> {
+        self._upsert(collection_name, points, true).await
+    }
+
+    #[inline]
+    async fn _upsert(
+        &mut self,
+        collection_name: impl ToString,
+        points: Vec<Point>,
+        block: bool,
+    ) -> Result<PointsOperationResponse> {
+        let result = self
+            .points_api
+            .upsert(UpsertPoints {
+                collection_name: collection_name.to_string(),
+                wait: Some(block),
+                points: points.into_iter().map(|p| p.into()).collect(),
+            })
+            .await?;
+
+        Ok(result.into_inner())
+    }
+}
+
+pub struct Point {
```

Comment:
It might cause confusion if we will use this custom point structure in some methods and original PointStruct in other, don't you think? Also why is it necessary to re-define such a basic data structure? It might cause much more effort to maintain the client, cause if we change the proto - we will also need to change all custom structures manually

Reply:
I built it because the id column in `PointStruct` has an awkward nested option in an option, and there isn't really a graceful way to have `PointStruct` hold the `Payload`. The idea was to build a layer between the generated proto files and the application using the lib.